### PR TITLE
feat: Add support for automated patching to the copr script

### PR DESCRIPTION
**Pull request by RKNF404**
Parallel PR with the submodule PR to add patching automation.

### CODE REVIEW

File: update-vanadium-patches.sh [hunk 1]
```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+gpwd=$(pwd)
+readonly path="./vanadium_patches/"
```
Comment (qoijjj) [inline update-vanadium-patches.sh:4]: can you name this something more descriptive?

File: update-vanadium-patches.sh [hunk 2]
```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+gpwd=$(pwd)
```
Comment (qoijjj) [inline update-vanadium-patches.sh:3]: and this

File: update-vanadium-patches.sh [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+smallpatches=() # truncated version of the patches for comparison
+vpatches=() # vanadium patches
+svpatches=() # truncated version for comparisons
+
+# gets the current patches
+get_patches() {
+	cd $path
+	patches=(*.patch)
+	for ((i=0; i<${#patches[@]}; i++)); do
+		smallpatches[$i]="${patches[$i]:4}"
```
Comment (qoijjj) [inline update-vanadium-patches.sh:16]: on the off chance vanadium changes their format, can we make the script fail if any of the patches don't start with four integers?

File: update-vanadium-patches.sh [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+	done
+	cd $gpwd
+}
+
+# gets Vanadium patches by cloning the repository temporarily
+get_vpatches() {
+	cd uvp-tmp/
```
Comment (qoijjj) [inline update-vanadium-patches.sh:23]: uvp?

File: update-vanadium-patches.sh [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	retry=0
+	while [ true ]; do
```
Comment (qoijjj) [inline update-vanadium-patches.sh:25]: how often does this fail to justify a retry?

File: update-vanadium-patches.sh [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+			fi
+			echo "Retrying..."
+			retry=$((retry+1))
+		else
+			break
+		fi
+	done
+	cd Vanadium/patches/
+	vpatches=(*.patch)
+	for ((i=0; i<${#vpatches[@]}; i++)); do
```
Comment (qoijjj) [inline update-vanadium-patches.sh:47]: very similar logic between this and get_patches. potential refactor?

File: update-vanadium-patches.sh [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+	cd $gpwd
+}
+
+update_patches() {
+	get_patches
+	get_vpatches
+	cd $path
+	counterU=0 # updated counter
+	counterR=0 # removed counter
+	counterNF=0 # not found counter, used to determine if it should be removed
```
Comment (qoijjj) [inline update-vanadium-patches.sh:59]: can you name these more descriptively?

File: update-vanadium-patches.sh [hunk 8]
```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+gpwd=$(pwd)
+readonly path="./vanadium_patches/"
+readonly url=https://raw.githubusercontent.com/GrapheneOS/Vanadium/main/patches
+patches=() # current patch names
+smallpatches=() # truncated version of the patches for comparison
+vpatches=() # vanadium patches
```
Comment (qoijjj) [inline update-vanadium-patches.sh:8]: can you name these such that the comments aren't needed? even if the variable names are very long. improves readability.

**Comment by qoijjj** [comment]
overall looks good, just some style comments. tyvm!

File: update-vanadium-patches.sh (hunk 5 above)
Comment (qoijjj) [reply update-vanadium-patches.sh:25]: @RKNF404 i meant in practice, how often does `git clone` from github fail? seems like it wouldn't need a retry

**Review by RoyalOughtness — APPROVED** [review]

**Pull request merged**